pandas/sparse/tests/test_array.py

```diff
@@ -159,6 +22,58 @@ def setUp(self):
         self.arr = SparseArray(self.arr_data)
         self.zarr = SparseArray([0, 0, 1, 2, 3, 0, 4, 5, 0, 6], fill_value=0)
 
+    def test_constructor_dtype(self):
+        arr = SparseArray([np.nan, 1, 2, np.nan])
+        self.assertEqual(arr.dtype, np.float64)
```

Comment:
I would test the fill_value on all of the these (e.g. nan as the default, 0 on next example)